api/client.go

```diff
@@ -79,6 +79,10 @@ type Client interface {
 	Do(context.Context, *http.Request) (*http.Response, []byte, error)
 }
 
+type CloseIdler interface {
```

Comment:
Ideally we can comment what this is for, but we can do in a separare PR, thanks!